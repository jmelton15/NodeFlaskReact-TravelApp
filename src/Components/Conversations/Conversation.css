.Conversation-AvatarPic {
    height:60px;
    width:60px;
    border-radius: 30px;
    margin-right:20px
}

.Conversation {
    margin-top: 25px;
    display: flex;
    align-items: center;
    margin-left: 20px;
}

.Conversation:hover {
    cursor: pointer;
}

@media only screen and (min-device-width: 5px) and (max-device-width: 420px) {
    .Conversation-AvatarPic {
        height:50px;
        width:50px;
        margin-right: 10px;
    }
    #message-body {
        margin-right: 10px;
    }
}