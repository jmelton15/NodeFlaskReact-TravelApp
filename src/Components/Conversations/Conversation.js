import sampleImg from "../../Images/fernTable.jpg";
import "./Conversation.css";

const Conversation = ({connection,getMessages}) => {
    return (
        <div className="Conversation" onClick={() => getMessages(connection.user_id)}>
            <img className="Conversation-AvatarPic" src={sampleImg} alt="Avatar Pic"/>
            <div id="message-body">
                <h5>{connection.username}</h5>
            </div>
        </div>
    )
}

export default Conversation;