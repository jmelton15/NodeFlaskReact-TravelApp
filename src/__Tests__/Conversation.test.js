import { fireEvent, render } from '@testing-library/react';
import {user,connection} from "./TestHelperObjects";
import Conversation from '../Components/Conversations/Conversation';

// simple smoke test for Conversation component
it("renders a Conversation component without crashing", function() {
    render(<Conversation connection={connection} getMessages={() => console.log("get messages")}/>);
});

// simple snapshot test for Conversation Component
it("should match snapshot for Conversation", function() {
    const {asFragment} = render (<Conversation connection={connection} getMessages={() => console.log("get messages")}/>);
    expect(asFragment()).toMatchSnapshot();
});
