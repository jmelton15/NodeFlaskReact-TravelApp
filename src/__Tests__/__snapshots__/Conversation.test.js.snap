// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot for Conversation 1`] = `
<DocumentFragment>
  <div
    class="Conversation"
  >
    <img
      alt="Avatar Pic"
      class="Conversation-AvatarPic"
      src="some pic route"
    />
    <div
      id="message-body"
    >
      <h5>
        TESTCONN
      </h5>
    </div>
  </div>
</DocumentFragment>
`;
